test/shared/src/main/scala/zio/test/TimeVariants.scala

```diff
@@ -16,13 +16,29 @@
 
 package zio.test
 
+import zio.ZIO
 import zio.duration.Duration
 import zio.random.Random
+import zio.stream.ZStream
 
-import java.time.{Instant, LocalDateTime, OffsetDateTime, ZoneOffset}
+import java.time._
 
 trait TimeVariants {
 
+  /**
+   * A generator of `java.time.DayOfWeek` values. Doesn't have any shrinking.
+   */
+  final def anyDayOfWeek: Gen[Random, DayOfWeek] =
+    Gen.oneOf(
```

Comment:
You can use `Gen.elements` here to avoid constructing another layer of generators.